src/ProgramSpeedController.js

```diff
@@ -0,0 +1,36 @@
+// @flow
+
+import React from 'react';
+import { Form } from 'react-bootstrap';
+import type { IntlShape } from 'react-intl';
+import { injectIntl } from 'react-intl';
+import { ReactComponent as SlowIcon } from './svg/Slow.svg';
+import { ReactComponent as FastIcon } from './svg/Fast.svg';
+import './ProgramSpeedController.scss';
+
+type ProgramSpeedControllerProps = {
+    intl: IntlShape,
+    onChange: (e: SyntheticInputEvent<HTMLInputElement>) => void
```

Comment:
I suggest moving the handling of the `HTMLInputElement` event to this component: give it a bit more responsibility.

- Add a property which is the list of values (in our case, App will provide the list `[2000, 1500, 1000, 500, 250]`)
- The `onChange` callback is called with one of the values provides from the list -- so it's called with 2000 for example, or 500, rather than the `HTMLInputElement`

So clients are isolated more from the implementation details of this component: they provide a list of values, and get notified when one of them has been selected.